### PR TITLE
Fix SurveyMonkey Android SDK manifest

### PR DESCRIPTION
**Pull request by kahakai**
Hello! According to the [Google Play's target API level requirement](https://developer.android.com/google/play/requirements/target-sdk),
> Starting in November 2022, app updates must target API level 31 or above and adjust for behavioral changes in Android 12.

Looking at the section [Migrate from Android 11 (API level 30) to Android 12 (API level 31)](https://developer.android.com/google/play/requirements/target-sdk#pre12), there is a point about Intent filters.
> Intent filters: If your app contains [activities](https://developer.android.com/guide/components/activities/intro-activities), [services](https://developer.android.com/guide/components/services), or [broadcast receivers](https://developer.android.com/guide/components/broadcasts) that use [intent filters](https://developer.android.com/guide/components/intents-filters#Receiving), you must explicitly declare the [android:exported](https://developer.android.com/guide/topics/manifest/activity-element#exported) attribute for these components.

I did explicitly declare the `android:exported` attribute for all the components of the app beforehand as it has been a best-practice for Android development and even recommended by Google in the documentation for long time.

If you set the `targetSdkVersion` to `31` then builds fail because of `react-native-survey-monkey`. You will see a generic message `Manifest merger failed with multiple errors, see logs`.
image
Running Gradle with the `--info` flag helps as it logs the task and manifest merger steps.
image
Here you go, the last one before a fail is `com.surveymonkey:surveymonkey-android-sdk:2.0.0`, specifically `com.surveymonkey.surveymonkeyandroidsdk.SMFeedbackActivity`.

The problem with the abandoned SurveyMonkey Mobile SDK is that the `android:exported` attribute is not declared in the manifest and probably never will, thus we have to add it by ourselves. We don't have the source code of the SDK because it is distributed as an [AAR](https://developer.android.com/studio/projects/android-library) (Android library), look at the [surveymonkey-android-sdk](https://github.com/SurveyMonkey/surveymonkey-android-sdk/blob/master/surveymonkey_android_sdk.aar) repository.

Luckily, we have an ability to manage Manifest files with build tools. All the manifest files are combined into one file. We can tweak the SurveyMonkey Android SDK manifest with the `merge` [node marker](https://developer.android.com/studio/build/manage-manifests#node_markers) to add the required `android:exported` attribute.

TLDR: I added the now required `android:exported` attribute to the SurveyMonkey Android SDK manifest.

### CODE REVIEW

**Comment by mikemonaco** [comment]
Thanks for this fix, helped us out! I also was having problems with 

`implementation 'com.surveymonkey:surveymonkey-android-sdk:2.0.0'`

I was getting the following error

`Could not find com.surveymonkey:surveymonkey-android-sdk:2.0.0`

I ended up making my own fork which bundles the aar file directly as well as implementing your fix: https://github.com/mikemonaco/react-native-survey-monkey/commit/8999ec7bcbaa9daca1fc335defb517928a7923d3

**Comment by rangerscience** [comment]
It looks like maybe @yarikpwnzer is no longer active on Github, but @rMozes is. What's needed to get this merged? :)
